cleanup: ensuring header levels do not skip

dandhlee opened the pull request:
Cleaned any of the remaining h4/h5/h6 elements that didn't need to appear as headers, but simply needed to appear as \<p> elements.

Any headers that appeared at the top of the pages as h4/h5/h6 were changed to be h2 elements, to ensure we go from h1 title to h2 headers then h2/h3 body headers.

In doing so some headers changed the way they look, for example

```
Assembly: abc.dll
```
changed to
```
Assembly
abc.dll
```

as `Assembly` would appear as h2 header but not `abc.dll`.

Fixes #159

dandhlee committed:
fixing Examples to Example

dandhlee committed:
Changing examples from h5 to normal heading

dandhlee committed:
Eliminating H2 grouping and moving one levels up

dandhlee committed:
getting rid of additional h2/h3 headers

dandhlee committed:
Adding <strong> tags to stripped headers

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into header_level

dandhlee committed:
Updating tokens with h5 headers

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into plural_headers

dandhlee committed:
cleanup: ensuring headers do not skip levels

tbpg commented on third_party/docfx/templates/devsite/partials/class.header.tmpl.partial:
```diff
@@ -4,7 +4,7 @@
 {{#inClass}}
 <div class="inheritance">
-  <h5>{{__global.inheritance}}</h5>
+  <h2>{{__global.inheritance}}</h2>
```
I don't have a good sense of what should be a header and what should be a paragraph. This seems alright, but Assembly as a header seems out of place.

dandhlee replied:
After chatting offline, keeping them as `h2` would makes sense for *strict* accessibility guideline, however it's somewhat loosely applied as we're not using headers for aesthetic purposes but for organization and whether to keep them for TOC or not.

As they are not quite member labels and since they were `h5`/`h6` previously, it is suggested to use `h3` as per the UX suggestion. It would slightly go against the accessibility guidelines.

`<p>` is not a good replacement for headers as it "denotes a paragraph, and headers should be marked up as an actual header".

Whether we keep these as `h2` or `h3`, we have the option to prune them out of the right TOC by adding `class="hide-from-toc"` to the headers.

tbpg reviewed (changes requested):
Chatting offline about header vs paragraph for these.

dandhlee commented:
Moving on as-is, we'll experiment with seeing how people feel with the `h2` headers.

dandhlee committed:
Merge branch 'master' into flatten_levels

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into flatten_levels

tbpg commented on testdata/goldens/dotnet/Google.Cloud.Asset.V1.AnalyzeIamPolicyResponse.Types.IamPolicyAnalysis.Types.Stats.Types.NodeType.html:
```diff
@@ -14,9 +14,11 @@ <h1 class="page-title">Enum AnalyzeIamPolicyResponse.Types.IamPolicyAnalysis.Typ
+  <a class="xref" href="Google.Cloud.Asset.V1.html">Google.Cloud.Asset.V1</a>
+  <h2>Assembly</h2>
+  Google.Cloud.Asset.V1.dll
+  <h2>Syntax</h2>
```
I wonder if we should make this "Syntax" the same as other syntax nodes -- no "Syntax" header and just stick the code block right under the `h1`.

dandhlee replied:
I think that sounds better, also helps with consistency.

tbpg commented on testdata/goldens/dotnet/Google.Cloud.Asset.V1.AnalyzeIamPolicyResponse.Types.IamPolicyAnalysis.Types.Stats.Types.NodeType.html:
```diff
@@ -14,9 +14,11 @@ <h1 class="page-title">Enum AnalyzeIamPolicyResponse.Types.IamPolicyAnalysis.Typ
-  <h5 id="Google_Cloud_Asset_V1_AnalyzeIamPolicyResponse_Types_IamPolicyAnalysis_Types_Stats_Types_NodeType_syntax">Syntax</h5>
+  <h2>Namespace</h2>
+  <a class="xref" href="Google.Cloud.Asset.V1.html">Google.Cloud.Asset.V1</a>
+  <h2>Assembly</h2>
+  Google.Cloud.Asset.V1.dll
```
@jskeet should we keep Assembly?

jskeet replied:
Yes, I'd expect to see that.

tbpg replied:
@dandhlee I think `Google.Cloud.Asset.V1.dll` needs to be in a `<p>`?

dandhlee replied:
done.

tbpg commented on third_party/docfx/templates/devsite/partials/uref/parameters.tmpl.partial:
```diff
@@ -1,7 +1,7 @@
 
 {{#properties.0}}
-<h6 class="parameters">Properties</h6>
+<strong>Properties</strong>
```
I don't think this should just be `strong`?

dandhlee replied:
The header above it would likely be `<strong>{{__global.parameters}}</strong>`, if not then `h3` above, so it would have to be h4 at best. I think it's fine to keep it as strong.

tbpg commented on third_party/docfx/templates/devsite/partials/uref/namespace.examples.tmpl.partial:
```diff
@@ -10,7 +10,7 @@
 {{#name}}
 {{^name.0}}
-<h6 class="notranslate">{{name}}</h6>
+<div class="notranslate"><strong>{{name}}</strong></div>
```
Semantically, I think this should be a header?

dandhlee replied:
It doesn't quite align with the rest of the TOC on right nav that it shows up as a header. I think it's fine to leave it out, currently it only affects the goldens for Go however.

tbpg commented on third_party/docfx/templates/devsite/partials/uref/class.tmpl.partial:
```diff
@@ -197,7 +197,11 @@
 <div class="markdown level1 remarks">{{{remarks}}}</div>
 {{/remarks}}
 {{#example.0}}
+<<<<<<< HEAD
```
Git merge issue.

tbpg commented on third_party/docfx/templates/devsite/partials/namespace.tmpl.partial:
```diff
@@ -5,9 +5,8 @@
 <div class="markdown level0 conceptual">{{{conceptual}}}</div>
 <div class="markdown level0 remarks">{{{remarks}}}</div>
 {{#children}}
-  <h2 id="{{id}}">{{>partials/namespaceSubtitle}}</h2>
```
Is this supposed to be deleted?

dandhlee replied:
forgot to catch it, will revert it.

tbpg commented on third_party/docfx/templates/devsite/partials/class.tmpl.partial:
```diff
@@ -2,7 +2,6 @@
 
 {{>partials/class.header}}
 {{#children}}
-<h2 id="{{id}}">{{>partials/classSubtitle}}</h2>
```
Should this be deleted?

dandhlee replied:
will revert it.

dandhlee committed:
Making additional updates

dandhlee commented:
Made some changes, PTAL

tbpg commented on testdata/goldens/go/index.html:
```diff
@@ -783,7 +783,7 @@ <h2 id="cloud_google_com_go_storage_BucketHandle" data-uid="cloud.google.com/go/
 </div>
         <div class="markdown level1 conceptual"></div>
         <strong>Examples</strong>
-        <h6 class="notranslate">exists</h6>
+        <div class="notranslate">exists</div>
```
This doesn't look right to me. It seems like for this page, `Examples` should be an `h3` since it's under the `h2` `BucketHandle`. Then `exists` should be an `h4`, right?

If someone is using a screen reader, they should be able to jump to the `Examples` header, and to each example name within `Examples`.

dandhlee replied:
Makes sense. Modified Examples and some sub-headers to have `h3` and above headers.

dandhlee committed:
Modifying Examples and sub-header sizes

dandhlee commented:
Modified assembly to be inside `<p>` and updated Example and its subheaders to
1) be a header
2) have appropriate organization and sized header accordingly

PTAL again!

dandhlee committed:
Merge branch 'master' into flatten_levels

dandhlee committed:
Add updated java goldens

tbpg commented on testdata/goldens/java/com.google.cloud.speech.v1.LongRunningRecognizeMetadata.Builder.html:
```diff
@@ -221,12 +224,10 @@ <h5>Inherited Members</h5>
-  <div class="codewrapper">
-    <pre><code class="prettyprint">public static final class LongRunningRecognizeMetadata.Builder extends GeneratedMessageV3.Builder&lt;LongRunningRecognizeMetadata.Builder&gt; implements LongRunningRecognizeMetadataOrBuilder</code></pre>
-  </div>
+  <h2>Namespace</h2>
+  
+  <h2>Assembly</h2>
+  <p>.dll</p>
```
Namespace and Assembly should not be included if there isn't a corresponding value.

dandhlee replied:
Namespace is updated in #173

dandhlee replied:
Updated!

tbpg reviewed (changes requested):
One more comment. Please stage and share a link internally to confirm this looks good before we merge.

dandhlee committed:
Merge branch 'master' into flatten_levels

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into flatten_levels

dandhlee committed:
revert <p> for assembly

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into flatten_levels

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into flatten_levels

tbpg reviewed (approved):
Thank you!

dandhlee committed:
Merge branch 'master' of github.com:googleapis/doc-templates into flatten_levels
